plutus-ledger-api/src/Plutus/ApiCommon.hs

```diff
@@ -343,3 +347,52 @@ evaluateScriptCounting lv pv verbose ectx p args = swap $ runWriter @LogOutput $
     tell logs
     liftEither $ first CekError $ void res
     pure final
+
+
+{-| A valid parameter name has to be enumeration, bounded, ordered, and
+prettyprintable in a "lowerKebab" way.
+
+Each API version should expose such an enumeration as an ADT and create
+an instance of ParamName out of it.
+-}
+class (Enum a, Ord a, Bounded a) => IsParamName a where
+   showParamName :: a -> String
+   default showParamName :: (Generic a, GIsParamName (Rep a)) => a -> String
```

Comment:
FWIW I think the "modern" way of doing this thing is with a newtype and deriving via, i.e.
```
newtype GenericParamName a = GenericParamName a
instance (Generic a, GIsParamName (Rep a)) => IsParamName a where ...

data Foo ...
    deriving IsParamName via (GenericParamName Foo)
```

Reply:
I will try it out. I am old-school